src/System.Diagnostics.PerformanceCounter/tests/PerformanceCounterCategoryTests.cs

```diff
@@ -0,0 +1,368 @@
+            Assert.Equal(".", pcc.MachineName);
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_CreatePerformanceCounterCategory_NullTests()
+        {
+            Assert.Throws<ArgumentNullException>(() => new PerformanceCounterCategory(null, "."));
+            Assert.Throws<ArgumentException>(() => new PerformanceCounterCategory(string.Empty, "."));
+            Assert.Throws<ArgumentException>(() => new PerformanceCounterCategory("category", string.Empty));
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_SetCategoryName_Valid()
+        {
+            PerformanceCounterCategory pcc = new PerformanceCounterCategory();
+            pcc.CategoryName = "Processor";
+            Assert.Equal("Processor", pcc.CategoryName);
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_SetCategoryName_Invalid()
+        {
+            PerformanceCounterCategory pcc = new PerformanceCounterCategory();
+
+            Assert.Throws<ArgumentNullException>(() => pcc.CategoryName = null);
+            Assert.Throws<ArgumentException>(() => pcc.CategoryName = string.Empty);
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_SetMachineName_Invalid()
+        {
+            PerformanceCounterCategory pcc = new PerformanceCounterCategory();
+
+            Assert.Throws<ArgumentException>(() => pcc.MachineName = string.Empty);
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_SetMachineName_ValidCategoryNameNull()
+        {
+            PerformanceCounterCategory pcc = new PerformanceCounterCategory();
+
+            pcc.MachineName = "machineName";
+            Assert.Equal("machineName", pcc.MachineName);
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_SetMachineName_ValidCategoryNameNotNull()
+        {
+            PerformanceCounterCategory pcc = new PerformanceCounterCategory();
+
+            pcc.CategoryName = "Processor";
+            pcc.MachineName = "machineName";
+            Assert.Equal("machineName", pcc.MachineName);
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_GetCounterHelp()
+        {
+            var name = nameof(PerformanceCounterCategory_GetCounterHelp) + "_Counter";
+            var category = name + "_Category";
+
+            if (PerformanceCounterCategory.Exists(category))
```

Comment:
This category exists/delete pattern you are using all over, you could make it into a private helper function

Reply:
Ah you have one: DeleteCategory()